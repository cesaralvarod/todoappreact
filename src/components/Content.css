.content {
  display: flex;
  flex-direction: column;
  margin-top: 3rem;
  gap: 3rem;
}

@media (min-width: 700px) {
  .content {
    flex-direction: row;
  }
}
