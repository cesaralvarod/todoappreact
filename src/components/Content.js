import React, { useReducer } from "react";

// Components

import TasksContent from "./TasksContent";
import FormContent from "./FormContent";

// Styles

import "./Content.css";

const Content = () => {
  console.log("cargado");

  let initialArrData = { data: [], total: 0 };

  const arrDataReducer = (state, action) => {
    switch (action.type) {
      case "add":
        return {
          data: [...state.data, action.dataForm],
          total: state.total + 1,
        };
      default:
        throw new Error("Input requires an action type");
    }
  };

  const [arrData, setArrData] = useReducer(arrDataReducer, initialArrData);

  return (
    <div className="content">
      <TasksContent arrData={arrData} />
      <FormContent setArrData={setArrData} />
    </div>
  );
};

export default Content;
