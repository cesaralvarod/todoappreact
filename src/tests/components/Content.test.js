import React from "react";
import { shallow } from "enzyme";
import Content from "../../components/Content";

describe("Testing in component Content", () => {
  test("Component Content should show correctly", () => {
    const wrapper = shallow(<Content />);

    console.log(wrapper.html());

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find("TasksContent").exists()).toBeTruthy();
  });
});
