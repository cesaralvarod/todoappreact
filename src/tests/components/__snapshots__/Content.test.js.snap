// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in component Content Component Content should show correctly 1`] = `
<div
  className="content"
>
  <ContextProvider
    value={
      Object {
        "arrData": Object {
          "data": Array [],
        },
        "setArrData": [Function],
      }
    }
  >
    <TasksContent />
    <Memo() />
  </ContextProvider>
</div>
`;
